### PR TITLE
Add autosubmit bot support for dependabot.

### PR DESCRIPTION
**Pull request by godofredoc**
Dependabot uses the author association instead of login to identify the
author. This changes explicitly checks for dependabot PRs.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart [hunk 1]
```diff
@@ -238,13 +238,15 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
...
       final String? author = pullRequest['author']['login'] as String?;
+      final String? authorAssociation = pullRequest['authorAssociation'] as String;
       final String id = pullRequest['id'] as String;
       final String repoFullName = pullRequest['baseRepository']['nameWithOwner'] as String;
       final RepositorySlug slug = RepositorySlug.full(repoFullName);
       final String title = pullRequest['title'] as String;
 
       final Set<String?> changeRequestAuthors = <String?>{};
-      final bool hasApproval = config.rollerAccounts.contains(author) ||
+      final bool hasApproval = authorAssociation == 'dependabot[bot]' ||
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart:248]: Can you add a test?

File: app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart [hunk 2]
```diff
@@ -238,13 +238,15 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
...
+          config.rollerAccounts.contains(author) ||
           _checkApproval(
             author,
             pullRequest['authorAssociation'] as String,
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart:252]:
```suggestion
            authorAssociation,
```

File: app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart [hunk 3]
```diff
@@ -238,13 +238,15 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
         continue;
       }
       final String? author = pullRequest['author']['login'] as String?;
+      final String? authorAssociation = pullRequest['authorAssociation'] as String;
```
Comment (CaseyHillers) [inline app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart:241]:
```suggestion
      final String authorAssociation = pullRequest['authorAssociation'] as String;
```

**Review by jmagman — COMMENTED** [review]
Can you also get rid this line from my (failed) attempt to do this?
https://github.com/flutter/cocoon/blob/f2b31264e627e6f33ec29fac3b6bfb39e305285c/app_dart/lib/src/service/config.dart#L290

File: app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart (hunk 2 above)
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart:252]: done!

File: app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart (hunk 3 above)
Comment (godofredoc) [reply app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart:241]: done

**Review by jmagman — APPROVED** [review]
LGTM

**Review by CaseyHillers — APPROVED** [review]

**Pull request merged**